File: client/src/app/file-system-entry/file-system-entry.ts
export class FileSystemEntry {

    public name: string;
    public type: 'file' | 'directory';
    public path: string;
    public entries: FileSystemEntry[];

    constructor(name: string, type: 'file' | 'directory', path: string) {
        this.name = name;
        this.type = type;
        this.path = path;
    }

}

File: server/file_system_entry.py
class FileSystemEntry():
    Name = ''
    Type = ''
    Path = ''

    def __init__(self, name, type, path):
        self.Name = name
        self.Type = type
        self.Path = path

    def __iter__(self):
        yield 'name', self.Name
        yield 'type', self.Type
        yield 'path', self.Path